community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -265,10 +299,33 @@ protected void assertPagedFileStillMapped()
 
     // --- IO methods:
 
+    private void checkBounds( int position )
+    {
+        if ( position > size )
+        {
+            throw new IndexOutOfBoundsException( getOutOfBoundsMessage( position ) );
+        }
+    }
+
+    private String getOutOfBoundsMessage( int position )
+    {
+        if ( size > 0 )
+        {
+            return "Position " + position + " is greater than the upper " +
+                   "page size bound of " + size;
+        }
+        else
+        {
+            return "The PageCursor is not bound to a page. " +
+                   "Maybe next() returned false or was not called, or the cursor has been closed.";
+        }
+    }
+
     @Override
     public final byte getByte()
     {
-        byte b = page.getByte( offset );
+        checkBounds( offset + 1 );
```

Comment:
I assume you mean something like `Bytes.SIZE_OF_BYTE`, because otherwise I find `Byte.BYTES` to be a confusing constant. Either way, I don't think it really adds anything in this case.